k8s/k8s.bzl

```diff
@@ -12,38 +12,46 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Rules for manipulation of K8s constructs."""
+
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
 load(":with-defaults.bzl", "k8s_defaults")
 load("//toolchains/kubectl:kubectl_configure.bzl", "kubectl_configure")
 
-def k8s_repositories():
-  """Download dependencies of k8s rules."""
+def k8s_repositories(build_kubectl_srcs = False):
+    """Download dependencies of k8s rules."""
 
-  # Used by utilities for roundtripping yaml.
-  http_archive(
-    name = "yaml",
-    build_file_content = """
+    # Used by utilities for roundtripping yaml.
+    http_archive(
+        name = "yaml",
+        build_file_content = """
 py_library(
     name = "yaml",
     srcs = glob(["*.py"]),
     visibility = ["//visibility:public"],
 )""",
-    sha256 = "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab",
-    urls = [("https://pypi.python.org/packages/4a/85/" +
-           "db5a2df477072b2902b0eb892feb37d88ac635d36245a72a6a69b23b383a" +
-           "/PyYAML-3.12.tar.gz")],
-    strip_prefix = "PyYAML-3.12/lib/yaml",
-  )
+        sha256 = "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab",
+        urls = [("https://pypi.python.org/packages/4a/85/" +
+                 "db5a2df477072b2902b0eb892feb37d88ac635d36245a72a6a69b23b383a" +
+                 "/PyYAML-3.12.tar.gz")],
+        strip_prefix = "PyYAML-3.12/lib/yaml",
+    )
 
-  # Register the default kubectl toolchain targets for supported platforms
-  # note these work with the autoconfigured toolchain
-  native.register_toolchains(
-    "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_linux_toolchain",
-    "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_osx_toolchain",
-    "@io_bazel_rules_k8s//toolchains/kubectl:kubectl_windows_toolchain",
-  )
+    # Register the default kubectl toolchain targets for supported platforms
+    # note these work with the autoconfigured toolchain
+    native.register_toolchains(
```

Comment:
Wondering if we should move the toolchain registration to be under the "k8s_config" if condition as well? Allows users to decide their own toolchain registration priority order when defining their own toolchain configuration

Reply:
i think it makes sense, will do in an upcoming PR